src/entry_point.py

```diff
@@ -201,6 +256,311 @@ def _constructor_menuinst(prefix, pkg_names=None, root_prefix=None, remove=False
+                    continue
+                rc_content = target_path.read_text()
+                if shell == "powershell":
+                    pattern = CONDA_INITIALIZE_PS_RE_BLOCK
+                else:
+                    pattern = CONDA_INITIALIZE_RE_BLOCK
+                flags = re.MULTILINE
+                matches = re.findall(pattern, rc_content, flags=flags)
+                if not matches:
+                    continue
+                for prefix in prefixes:
+                    # Ignore .exe suffix to make the logic simpler
+                    if shell in ("csh", "tcsh") and sys.platform != "win32":
+                        sentinel_str = str(prefix / "etc" / "profile.d" / "conda.csh")
+                    else:
+                        sentinel_str = str(prefix / BIN_DIRECTORY / "conda")
+                    if sys.platform == "win32" and shell != "powershell":
+                        sentinel_str = win_path_to_unix(sentinel_str)
+                        # Remove /cygdrive to make the path shell-independent
+                        if sentinel_str.startswith("/cygdrive"):
+                            sentinel_str = sentinel_str[9:]
+                    if any(sentinel_str in match for match in matches):
+                        reverse_plan.append(initializer)
+                        break
+    return reverse_plan
+
+
+def _constructor_uninstall_subcommand(
+    uninstall_dir: str,
+    conda_clean: bool = False,
+    remove_condarcs: str | None = None,
+    remove_caches: bool = False,
+):
+    """
+    Remove a conda prefix or a directory containing conda environments.
+
+    This command also provides options to remove various cache and configuration
+    files to fully remove a conda installation.
+    """
+    from conda.base.constants import PREFIX_MAGIC_FILE
+
+    ENVS_DIR_MAGIC_FILE = ".conda_envs_dir_test"
+
+    uninstall_prefix = Path(uninstall_dir).expanduser().resolve()
+    if (
+        not (uninstall_prefix / PREFIX_MAGIC_FILE).exists()
+        and not (uninstall_prefix / ENVS_DIR_MAGIC_FILE).exists()
+    ):
+        raise OSError(
+            f"{uninstall_prefix} is not a valid conda environment or environments directory."
+        )
+
+    from shutil import rmtree
+
+    from conda.base.context import context, reset_context
+    from conda.cli.main import main as conda_main
+    from conda.core.initialize import print_plan_results, run_plan, run_plan_elevated
+
+    def _remove_file_directory(file: Path):
+        """
+        Try to remove a file or directory.
+
+        If the file is a link, just unlink, do not remove the target.
+        """
+        try:
+            if not file.exists():
+                return
+            if file.is_symlink() or file.is_file():
+                file.unlink()
+            elif file.is_dir():
+                rmtree(file)
+        except PermissionError:
+            pass
+
+    def _remove_file_and_parents(file: Path, stop_at: Path | None = None):
```

Comment:
This can be quite a destructive function, so I'd recommend adding separate tests and making it a global function.

For example, we need to ensure that `file` is contained in `stop_at`, right?

Reply:
I added a check, which actually made me find a bug. This made me add a warning to the README file that support for softlinks is limited.

The problem with writing separate tests is that we would have to create an entry point for it, which would expose that destructive function even more. I added another tests to ensure that the function does not a non-empty directory.

Reply:
Hm, cannot we import the `entry_point.py` module in the tests directly and target that function?

Reply:
This function is obsolete now that I don't aggressively remove parents anymore. So, no tests needed!